### DISCUSSION TITLE
Creating a new token

### DISCUSSION THREAD

**Discussion by lulutche** [markedjs/marked#2042] [Q&A]
Hi guys ! I'm actually trying to retrieve some metadata in a Github Flavor Markdown file. So i tried to create a token named "color" with that type of structure "[//]: # (metadata)". But actually my tokenizer don't recognize this like my new token but as a "link" token.Here is my code and the logs when i run this.

```
const marked = require ('marked');

const tokenizer = {
    color(src) {
        const match = src.match("(\[\/\/]: #) \(\w*\)");
        if (match) {
            return {
                type: 'color',
                text: match[0],
                raw: match[0]
            }
        }
        return false;
    }
}


marked.use({tokenizer});

const md = "
  [color]:<red>
  [test]:(test)
  
  [//]: # (red)
  
";

const tokens = marked.lexer(md);
console.log(tokens);

const html = marked.parser(tokens);
console.log(html);
console.log("Here it recognize as a link token instead of my color token");

src = "[//]: # (red)";
match = src.match(/\[\/\/]: # \([a-z]*\)/);
console.log(match[0]);
console.log("Here my regExp is right and recognize this structure");
```

![log_TokenizerTest](https://user-images.githubusercontent.com/80798744/117437057-fdb8c300-af2f-11eb-885d-6cd8588b59d6.png)

What am i doing wrong ?
Thanks for your help.
Lulutche

**Comment by UziTech** [comment]
There is no `color` tokenizer. The list of tokenizers can be found in the [docs](https://marked.js.org/using_pro#block-level-tokenizer-methods). There is a proposal for a way to add tokenizers (#1872) but it isn't finished yet.
